core/drivers/xiphera_trng.c

```diff
@@ -85,6 +82,12 @@ TEE_Result hw_get_random_bytes(void *buf, size_t len)
 	while (buffer_pos < len) {
 		uint32_t exceptions = cpu_spin_lock_xsave(&trng_lock);
 
+		/* Wait until we have value available */
+		if (!xiphera_trng_random_available()) {
+			cpu_spin_unlock_xrestore(&trng_lock, exceptions);
```

Comment:
I'd prefer matching each lock with a single unlock when possible to make it easier to read the code.

Reply:
Ok. I'll see how that could look.

Reply:
This is now changed. Actually dropped the whole FIFO construct at the same time as it simplified it quite a bit with this construct.

Originally we were advised by TRNG vendor not to cache previous results if everything is not consumed. So this returns the original approach.